httpbakery/agent/agent_test.go

```diff
@@ -117,6 +121,36 @@ func (s *agentSuite) TestSetUpAuth(c *gc.C) {
 	c.Assert(err, gc.Equals, nil)
 }
 
+func (s *agentSuite) TestAuthInfoFromEnvironment(c *gc.C) {
+	defer os.Setenv("BAKERY_AGENT_FILE", "")
+
+	f, err := ioutil.TempFile("", "")
+	c.Assert(err, gc.Equals, nil)
+	defer os.Remove(f.Name())
+	defer f.Close()
+
+	authInfo := &agent.AuthInfo{
+		Key: bakery.MustGenerateKey(),
+		Agents: []agent.Agent{{
+			URL:      "https://0.1.2.3/x",
+			Username: "bob",
+		}, {
+			URL:      "https://0.2.3.4",
+			Username: "charlie",
+		}},
+	}
+	data, err := json.Marshal(authInfo)
+	_, err = f.Write(data)
+	c.Assert(err, gc.Equals, nil)
+	f.Close()
+
+	os.Setenv("BAKERY_AGENT_FILE", f.Name())
+
+	authInfo1, err := agent.AuthInfoFromEnvironment()
+	c.Assert(err, gc.Equals, nil)
+	c.Assert(authInfo1, jc.DeepEquals, authInfo)
+}
```

Comment:
Please also add a test for the error when the environment variable isn't set.

Reply:
done.